operators/pkg/tenant-controller/tenant_controller.go

```diff
@@ -144,27 +154,87 @@ func (r *TenantReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	}
 
+
+
+	keepNsOpen := true; // keepNsOpen defines if we should close the personal namespace based on the last login date
+	                    // must be initialized for later use
+
+	// If the personalNamespace is not created and the lastLogin is blank, then this is the first time
+	// the tenant has logged in; update the lastLogin date to current time
+	if tn.Status.PersonalNamespace.Created==false && tn.Spec.LastLogin=="" {
+
+	  tn.Spec.LastLogin = time.Now().Format(time.RFC3339)
+      klog.Infof("First login of %s: setting lastLogin value of %s in tenant spec.", tn.Name, tn.Spec.LastLogin)
+
+	} else { // Otherwise, this is not the first time the user has logged in
+
+	  // So we check to see if last login was more than TENANT_WORKSPACE_KEEP_ALIVE_SECONDS in the past:
+	  // if so, temporarily delete the namespace
+      
+	  t, err := time.Parse(time.RFC3339, tn.Spec.LastLogin) // Read lastLogin string and converte to datetime
+	  if err != nil {
+		  klog.Errorf("Error in parsing lastLogin date in RFC3339 format of tenant %s -> %s", tn.Name, err)
+		  return ctrl.Result{}, retrigErr
+	  }
+
+      // Calculate time elapsed since lastLogin (now minus lastLogin in seconds)
+	  sPassed := int(time.Now().Sub(t).Seconds());
+
+	  klog.Infof("Last login of tenant %s was %d seconds ago", tn.Name, sPassed)
+
+	  if(sPassed>TENANT_WORKSPACE_KEEP_ALIVE_SECONDS) { // seconds
+	  	klog.Infof("Over %d seconds elapsed since last login of tenant %s: attempting to delete tenant namespace if not already deleted", TENANT_WORKSPACE_KEEP_ALIVE_SECONDS, tn.Name)
+	  	keepNsOpen = false;
+	  } else {
+	  	klog.Infof("Under %d seconds (limit) elapsed since last login of tenant %s: namespace is left as-is", TENANT_WORKSPACE_KEEP_ALIVE_SECONDS, tn.Name)
+	  }
+
+	}
+
+
+    nsOk := false; // nsOk must be initialized for later use
+
 	// update resource quota in the status of the tenant after checking validity of workspaces.
 	tn.Status.Quota = forge.TenantResourceList(workspaces, tn.Spec.Quota)
 
 	nsName := fmt.Sprintf("tenant-%s", strings.ReplaceAll(tn.Name, ".", "-"))
-	nsOk, err := r.createOrUpdateClusterResources(ctx, &tn, nsName)
-	if nsOk {
-		klog.Infof("Namespace %s for tenant %s updated", nsName, tn.Name)
-		tn.Status.PersonalNamespace.Created = true
-		tn.Status.PersonalNamespace.Name = nsName
-		if err != nil {
-			klog.Errorf("Unable to update cluster resource of tenant %s -> %s", tn.Name, err)
+
+    if keepNsOpen {
+	    nsOk, err = r.createOrUpdateClusterResources(ctx, &tn, nsName)
+		if nsOk {
+			klog.Infof("Namespace %s for tenant %s updated", nsName, tn.Name)
+			tn.Status.PersonalNamespace.Created = true
+			tn.Status.PersonalNamespace.Name = nsName
+			if err != nil {
+				klog.Errorf("Unable to update cluster resource of tenant %s -> %s", tn.Name, err)
+				retrigErr = err
+				tnOpinternalErrors.WithLabelValues("tenant", "cluster-resources").Inc()
+			}
+			klog.Infof("Cluster resources for tenant %s updated", tn.Name)
+			nsOk = true;
```

Comment:
This line is useless, as `nsOk = true` by definition.

Reply:
Resolved in [2034701](https://github.com/netgroup-polito/CrownLabs/pull/821/commits/20347013c27d350f57d082f8e61204a4d8194c64)